### PR TITLE
Update: Add `ArrowFunctionExpression` support to `require-jsdoc` rule

### PR DESCRIPTION
**Pull request by gyandeeps**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ x ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What rule do you want to change?**

* Add support for `ArrowFunctionExpression` nodes for `require-jsdoc` rule.

**Does this change cause the rule to produce more or fewer warnings?**

It will produce more warnings only when the option if set to true. By default its false, so out of the box it will not produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

* New option
  * Name - `ArrowFunctionExpression`
  * Default - `false`

**Please provide some example code that this change will affect:**

**Valid**

```js
/**
 * It returns 10
 */
var foo = () => {
    return 10;
}

/**
 * It returns 10
 */
var foo = () => () => {
    return 10;
}

setTimeout(() => {}, 10);
```

**Invalid**

```js
var foo = () => {
    return 10;
}
```

**What does the rule currently do for this code?**

Currently, we dont check for arrow functions.

**What will the rule do after it's changed?**

It will only check for arrow functions which are assigned to a variable. It will not check for anonymous arrow functions.

### CODE REVIEW

File: docs/rules/require-jsdoc.md [hunk 1]
```diff
@@ -87,6 +93,13 @@ function foo() {
 /**
  * It returns 10
  */
+var foo = () => {
```
Comment (ilyavolodin) [inline docs/rules/require-jsdoc.md]: Could you provide an example with `@props and @returns` just to make it clear that full jsDoc syntax is supported.

**Comment by markelog** [comment]
Would miss:
```js
b = () => {}
```
Is that desirable?

**Comment by gyandeeps** [comment]
Good point @markelog . Personally I fine with no checking the AssignmentExpression for arrow functions.
But want to see what others think.

**Comment by not-an-aardvark** [comment]
I think it would be nice to check for AssignmentExpressions, but it's not required (i.e. this PR would still be useful without it).

**Comment by platinumazure** [comment]
I'm okay with adding AssignmentExpression support later if users seem interested in it.

**Comment by ilyavolodin** [comment]
I think `AssignmentExpression` would fall under the enhancement of supporting `FunctionExpression`. Those two go hand-in-hand in my opinion.

**Comment by nzakas** [comment]
Looks like we have three 👍s and, of course, @gyandeeps is the champion, so marking as accepted.

**Pull request merged**